File: src/tasks_java/Role.java
package tasks_java;

public enum Role {
    ADMIN("полный доступ"),
    MODERATOR("частичный доступ"),
    USER("ограниченный доступ");

    private String privileges;

    Role(String privileges) {
        this.privileges = privileges;
    }

    public String getPrivileges() {
        return this.privileges;
    }

}

File: src/tasks_kotlin/Role.kt
package tasks_kotlin

enum class Role(
    private val privilege: Int
) {
    ADMIN(1),
    MODERATOR(0),
    USER(-1);

    fun hasSamePrivilege(other: Role): String {
        if (this.privilege == other.privilege)
            return "Привилегии равны"
        else if (this.privilege > other.privilege)
            return "Первая привилегия приоритетнее"
        else
            return "Вторая привилегия приоритетнее"
    }

}